### PR TITLE
Add nodejs-urijs

### PR DESCRIPTION
**Pull request by ohadlevy**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.15
* [ ] 1.14
* [ ] 1.13

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by ohadlevy** [comment]
required for https://github.com/theforeman/foreman/pull/4584

File: comps/comps-foreman-fedora24.xml [hunk 1]
```diff
@@ -184,6 +185,7 @@
       <packagereq type="default">rubygem-fog-digitalocean-doc</packagereq>
       <packagereq type="default">rubygem-fog-doc</packagereq>
       <packagereq type="default">rubygem-fog-google-doc</packagereq>
+      <packagereq type="default">rubygem-fog-libvirt-doc</packagereq>
```
Comment (dLobatog) [inline comps/comps-foreman-fedora24.xml]: It should be `nodejs-urijs-doc` , but this one is missing too.

File: comps/comps-foreman-rhel7.xml [hunk 2]
```diff
@@ -199,6 +199,7 @@
       <packagereq type="default">nodejs-select2</packagereq>
       <packagereq type="default">nodejs-stats-webpack-plugin</packagereq>
       <packagereq type="default">nodejs-style-loader</packagereq>
+      <packagereq type="default">nodejs-urijs</packagereq>
```
Comment (dLobatog) [inline comps/comps-foreman-rhel7.xml]: `-doc` comps missing

File: nodejs-urijs/nodejs-urijs.spec [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# FIXME
+# Remove nodejs_symlink_deps if installing bundled module with NPM
+
+%global npm_name urijs
+%global enable_tests 0
```
Comment (dLobatog) [inline nodejs-urijs/nodejs-urijs.spec]: Should be OK to `enable_tests 1`, other packages have it set too

File: nodejs-urijs/nodejs-urijs.spec [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+%{nodejs_symlink_deps} --check
+#$CHECK
+%endif
+
+%files
+%{nodejs_sitelib}/%{npm_name}
+
+%doc LICENSE.txt
+%doc CHANGELOG.md
+%doc LICENSE.txt
```
Comment (dLobatog) [inline nodejs-urijs/nodejs-urijs.spec]: Not sure why `LICENSE.txt` shows up twice, npm2rpm bug?

File: nodejs-urijs/nodejs-urijs.spec [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+%global npm_name urijs
+%global enable_tests 0
+
+%{?nodejs_find_provides_and_requires}
+
+Name: nodejs-%{npm_name}
+Version: 1.18.10
+Release: 1%{?dist}
+Summary: URI
```
Comment (dLobatog) [inline nodejs-urijs/nodejs-urijs.spec]: From npmjs.org:

> URI.js is a javascript library for working with URLs

**Comment by ohadlevy** [comment]
@dLobatog I just used the script, I assume all of it is npm2rpm bugs?

File: comps/comps-foreman-rhel7.xml (hunk 2 above)
Comment (ohadlevy) [inline comps/comps-foreman-rhel7.xml]: @dLobatog should this be handled by the script?

**Comment by ohadlevy** [comment]
@dLobatog updated.

File: foreman/foreman.spec [hunk 6]
```diff
@@ -288,6 +288,8 @@ BuildRequires: npm(stats-webpack-plugin) < 1.0.0
 BuildRequires: npm(stats-webpack-plugin) >= 0.2.1
 BuildRequires: npm(style-loader) < 1.0.0
 BuildRequires: npm(style-loader) >= 0.13.1
+BuildRequires: npm(urijs) < 1.19.0
```
Comment (ohadlevy) [inline foreman/foreman.spec]: Should this be in a separate PR?

File: comps/comps-foreman-rhel7.xml (hunk 2 above)
Comment (dLobatog) [inline comps/comps-foreman-rhel7.xml]: Yup, did it not do it?

File: foreman/foreman.spec (hunk 6 above)
Comment (dLobatog) [inline foreman/foreman.spec]: Same PR is fine, I think...

**Comment by ohadlevy** [comment]
@dLobatog i think it failed because i did them both in the same pr? otherwise I dont understand the failure?

File: comps/comps-foreman-rhel7.xml (hunk 2 above)
Comment (ohadlevy) [inline comps/comps-foreman-rhel7.xml]: @dLobatog correct, I had to manually change quite a few things...

**Comment by ohadlevy** [comment]
@dLobatog yep, looking at http://koji.katello.org/kojifiles/work/tasks/8094/8094/root.log I can see the following error:
``` Error: No Package found for npm(urijs) ```

**Comment by ohadlevy** [comment]
@dLobatog pushed without foreman spec changes.

**Comment by dLobatog** [comment]
@ohadlevy Yeah, it seems that it's not using mockchain to build packages in order, feel free to submit on another PR.

**Comment by ohadlevy** [comment]
@dLobatog tests are passing.

**Comment by ohadlevy** [comment]
@ehelms mind having a look? this currently blocks nightly.

**Comment by ehelms** [comment]
@ohadlevy Looks fine, but there is one outstanding comment and the comps need updating. Is there a follow up PR that updates an existing package to require this one?

**Comment by ohadlevy** [comment]
@ehelms https://github.com/theforeman/foreman-packaging/pull/1694

**Comment by ehelms** [comment]
Thanks for the follow up PR. can you adjust this PR to include this package in the comps in the appropriate places?

**Comment by ohadlevy** [comment]
@ehelms I did it initially but the test failed hence I broke it to two prs.

**Comment by ehelms** [comment]
@ohadlevy Yup! But if you don't add this package to the comps it won't end up in the repository. So either this PR (ideally) or #1694 should include this package in the comps. e.g. -- https://github.com/ohadlevy/foreman-packaging/blob/4a2da423de4343c3dca2c4557f4f52fac579c87b/comps/comps-foreman-fedora24.xml#L38

**Comment by ohadlevy** [comment]
@ehelms updated.

**Comment by ohadlevy** [comment]
@ehelms anything else missing? Thanks

File: comps/comps-foreman-fedora24.xml [hunk 7]
```diff
@@ -187,6 +188,8 @@
...
+      <packagereq type="default">rubygem-fog-urijs-doc</packagereq>
```
Comment (ehelms) [inline comps/comps-foreman-fedora24.xml]: Copy/paste here -- rubygem-fog-urijs --> nodejs-urijs

File: comps/comps-foreman-rhel7.xml [hunk 8]
```diff
@@ -304,6 +305,7 @@
       <packagereq type="default">tfm-rubygem-fog-google-doc</packagereq>
       <packagereq type="default">tfm-rubygem-fog-json-doc</packagereq>
       <packagereq type="default">tfm-rubygem-fog-libvirt-doc</packagereq>
+      <packagereq type="default">tfm-rubygem-fog-urijs-doc</packagereq>
```
Comment (ehelms) [inline comps/comps-foreman-rhel7.xml]: Copy/paste here -- rubygem-fog-urijs --> nodejs-urijs

Comment (ohadlevy) [inline comps/comps-foreman-rhel7.xml]: updated - thanks for catching that.
